plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -603,18 +634,23 @@ public Optional<TopNApplicationResult<ConnectorTableHandle>> applyTopN(
     @Override
     public Optional<TableFunctionApplicationResult<ConnectorTableHandle>> applyTableFunction(ConnectorSession session, ConnectorTableFunctionHandle handle)
     {
-        if (!(handle instanceof QueryFunctionHandle)) {
+        if (!(handle instanceof QueryFunctionHandle || handle instanceof ProcedureFunctionHandle)) {
```

Comment:
nit:
I think it would be clearer to do:
```
if (handle instanceof QueryFunctionHandle queryFunctionHandle) {
  ...
else if (handle instanceof ProcedureFunctionHandle procedureFunctionHandle) {
...
}
else{
    return Optional.empty();
}
```